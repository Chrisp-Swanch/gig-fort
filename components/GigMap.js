import { useState, useMemo } from "react";
import {
  StyleSheet,
  Text,
  View,
  Pressable,
  Image,
  ImageBackground,
  TouchableOpacity,
} from "react-native";
import MapView from "react-native-maps";
import { Marker, Callout } from "react-native-maps";
import { mapStyle } from "../util/mapStyle";
import { useGigs } from "../hooks/useGigs";
import { AntDesign } from "@expo/vector-icons";

const GigMap = ({ navigation }) => {
  const [selectedDateMs, setSelectedDateMs] = useState(Date.now());
  const gigs = useGigs();

  //generates current date in format DD/MM/YYYY
  const selectedDateString = useMemo(() => {
    const date = new Date(selectedDateMs);
    const dateToString = date.toString().slice(0, 15);
    return dateToString; // returns in form 'Tue Dec 20 2022'
  }, [selectedDateMs]);

  //Filtering through gigs to return only current day's gigs
  const gigsToday = gigs.filter((gig) => {
    const gigDate1 = new Date(gig.dateAndTime.seconds * 1000);
    const gigDate2 = gigDate1.toString().slice(0, 15); //return form 'Tue Dec 20 2022'
    return gigDate2 === selectedDateString;
  });

  //increments date by amount
  const addDays = (amount) => {
    setSelectedDateMs((curr) => curr + 1000 * 60 * 60 * 24 * amount);
  };

  return (
    <View style={styles.container}>
      <Text style={styles.headerText}>{`Gigs on ${selectedDateString}`}</Text>

      <View style={styles.imageText}>
        <Text style={styles.subHeader}>Tap on</Text>
        <Image
          style={styles.image}
          source={require("../assets/Icon_Gold_48x48.png")}
        />
        <Text style={styles.subHeader}> to see gig info</Text>
      </View>

      <MapView
        initialRegion={{
          latitude: -41.29416,
          longitude: 174.77782,
          latitudeDelta: 0.03,
          longitudeDelta: 0.03,
        }}
        style={styles.map}
        customMapStyle={mapStyle}
      >
        {gigsToday.map((gig, i) => {
          let venueName;
          if (gig.venue.length > 12) {
            venueName = `${gig.venue.substring(0, 5)}...`;
          } else {
            venueName = gig.venue;
          }
          return (
            <Marker
              style={{ flexDirection: "column" }}
              key={i}
              coordinate={{
                latitude: gig.location.latitude,
                longitude: gig.location.longitude,
              }}
              image={require("../assets/Icon_Gold_48x48.png")}
              onPress={() => {
                navigation.navigate("GigDetails", {
                  venue: gig.venue,
                  date: selectedDateString,
                  gigName: gig.gigName,
                  image: gig.image,
                  blurb: gig.blurb,
                  isFree: gig.isFree,
                  genre: gig.genre,
                  dateAndTime: gig.dateAndTime.seconds,
                  tickets: gig.tickets,
                });
              }} 
            >
              <View style = {{position: 'absolute'}}><Text>{venueName}</Text></View>
                
              
            </Marker>
          );
        })}
      </MapView>

      <View style={styles.buttonOptions}>
        <TouchableOpacity onPress={() => addDays(-1)} style={styles.touchable}>
          <AntDesign name="caretleft" size={36} color="#778899" />
          <Text style={{ fontFamily: "Helvetica-Neue", color: "#778899" }}>
            Previous day
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => addDays(1)} style={styles.touchable}>
          <AntDesign name="caretright" size={36} color="#778899" />
          <Text style={{ fontFamily: "Helvetica-Neue", color: "#778899" }}>
            Next day
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
  },
  map: {
    height: 450,
    width: "100%",
    margin: 10,
  },
  gigInfo: {
    // backgroundColor: '#68912b',
    // marginTop:20
  },
  gigInfo_text: {
    color: "white",
    fontFamily: "Helvetica-Neue",
    fontSize: 12,
    paddingTop: 22,
  },
  gigInfo_text_genre: {
    color: "white",
    fontFamily: "Helvetica-Neue",
    transform: [{ translateY: -5 }],
  },
  headerText: {
    color: "black",
    fontSize: 25,
    marginTop: 5,
    fontFamily: "Sofia-Pro",
    fontStyle: "bold",
    marginBottom: 10,
    marginTop: 15,
  },
  callout: {
    width: "auto",
    height: "auto",
    backgroundColor: "azure",
  },
  buttonOptions: {
    flexDirection: "row",
    justifyContent: "space-around",
    width: "92%",
  },
  buttonOptionsText: {
    margin: 5,
  },
  image: {
    height: 20,
    width: 20,
    paddingBottom: 10,
    margin: 0,
  },
  imageText: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  touchable: {
    flexDirection: "column",
    alignItems: "center",
  },
  subHeader: {
    fontStyle: "italic",
    fontFamily: "Helvetica-Neue",
    color: "#778899",
  },
});

export default GigMap;
