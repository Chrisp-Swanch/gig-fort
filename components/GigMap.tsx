
import { FC } from "react";
import { useState, useMemo } from "react";
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
  Platform,
  Dimensions
} from "react-native";
import MapView from "react-native-maps";
import { Marker, Callout } from "react-native-maps";
import { mapStyle } from "../util/mapStyle";
import { useGigs } from "../hooks/useGigs";
import { AntDesign } from "@expo/vector-icons";
import { format } from "date-fns";
import { mapProps } from "../routes/homeStack";


type MapScreenNavgationProp = mapProps['navigation']

interface Props {
    navigation: MapScreenNavgationProp
}

const GigMap:FC<Props> = ({ navigation }):JSX.Element => {
  const [selectedDateMs, setSelectedDateMs] = useState<number>(Date.now());
  const gigs = useGigs();

  //generates current date in format DD/MM/YYYY
  const selectedDateString:string = useMemo(() => {
    const formattedDate = format(new Date(selectedDateMs),'EEE LLL do Y')
    return formattedDate
  }, [selectedDateMs]);

  const currentDay:string = useMemo(() => {
    const formattedDay = format(new Date(selectedDateMs),'EEEE')
    return formattedDay
  },[selectedDateMs])

  const currentWeek:string = useMemo(() => {
    const formattedDay = format(new Date(selectedDateMs),'LLLL do Y')
    return formattedDay
  },[selectedDateMs])

  //Filtering through gigs to return only current day's gigs
  const gigsToday = gigs.filter((gig) => {
    const formattedGigDate = format(new Date(gig.dateAndTime.seconds * 1000), 'EEE LLL do Y')
    return formattedGigDate === selectedDateString;
  });
  

  //increments date by amount
  const addDays = (amount:number):void => {
    setSelectedDateMs((curr) => curr + 1000 * 60 * 60 * 24 * amount);
  };


  return (
    <View style={styles.container}>
      <View testID="gigMapHeader" style={styles.headerText}>
        <Text style = {styles.headerText_main}>{currentDay}</Text>
        <Text style = {styles.headerText_sub}>{currentWeek}</Text>
      </View>

      <View style={styles.imageText}>
        <Text style={styles.subHeader}>Tap on the</Text>
        <Image
          style={styles.image}
          source={require("../assets/map-pin-new.png")}
        />
        <Text style={styles.subHeader}>icons on the map to see more gig info</Text>
      </View>

      <View style = {styles.mapContainer}>
      <MapView
        initialRegion={{
          latitude: -41.29416,
          longitude: 174.77782,
          latitudeDelta: 0.03,
          longitudeDelta: 0.03,
        }}
        style={styles.map}
        customMapStyle={mapStyle}
      >
        {gigsToday.map((gig, i) => {

          let venueName:string;
          if (gig.venue.length > 12) {
            venueName = `${gig.venue.substring(0, 12)}...`;
          } else {
            venueName = gig.venue;
          }
          return (
            <Marker
              key={i}
              coordinate={{
                latitude: gig.location.latitude,
                longitude: gig.location.longitude,
              }}
              icon={require("../assets/map-pin-50pc.png")}
              onPress={() => {
                navigation.navigate("GigDetails", {
                  venue: gig.venue,
                  gigName: gig.gigName,
                  image: gig.image,
                  blurb: gig.blurb,
                  isFree: gig.isFree,
                  genre: gig.genre,
                  dateAndTime: {...gig.dateAndTime},
                  tickets: gig.tickets,
                });
              }} 
            >
              <Text style = {styles.gigInfo_text}>{gig.genre}</Text>
            </Marker>
          );
        })}
      </MapView>
      </View>

      {/* <View style={styles.buttonOptions}>
        <TouchableOpacity onPress={() => addDays(-1)} style={styles.touchable}>
          <AntDesign name="caretleft" size={36} color="#778899" />
          <Text style={{ fontFamily: "Helvetica-Neue", color: "#778899" }}>
            Previous day
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => addDays(1)} style={styles.touchable}>
          <AntDesign name="caretright" size={36} color="#778899" />
          <Text style={{ fontFamily: "Helvetica-Neue", color: "#778899" }}>
            Next day
          </Text>
        </TouchableOpacity>
      </View> */}

    </View>
  );
};

const {width:screenWidth, height:screenHeight} = Dimensions.get('window')
const mapWidth = screenWidth * 0.9 //this sets width to 90%
const mapHeight = mapWidth /0.91 //this set height  based on the figma map aspect ratio of 0.91



const styles = StyleSheet.create({
  container: {
    // flexDirection: "column",
    // alignItems: "center",
    flex:1
  },
  map: {
    height: '100%',
    width: '100%'
  },
  mapContainer:{
    marginTop: '10%',
    marginHorizontal: 20,
    width: mapWidth,
    height: mapHeight,
    ...Platform.select({
      ios: {
        borderRadius:26,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.3,
        shadowRadius: 2,
      },
      android: {
        overflow: 'hidden',
        borderRadius:26,
        elevation: 4,
      }
    })
  },
  gigInfo: {
    // backgroundColor: '#68912b',
    // marginTop:20
  },
  gigInfo_text: {
    fontSize:10,
    color:'black',
    fontFamily:'NunitoSans',
    paddingTop: 25,
    textAlign:'center',
    fontWeight:'bold'
  },
  gigInfo_text_genre: {
    color: "white",
    fontFamily: "Helvetica-Neue",
    transform: [{ translateY: -5 }],
  },
  headerText: {
    color: "black",
    fontSize: 25,
    marginTop: '0%',
    marginLeft: '7%',
    fontFamily: "NunitoSans",
    marginBottom: 10,
  },
  headerText_main: {
    fontFamily: "NunitoSans",
    fontSize:25,
    lineHeight:34.1
  },
  headerText_sub: {
    fontFamily:'LatoRegular',
    size:14,
    lineHeight:16.8
  },
  callout: {
    width: "auto",
    height: "auto",
    backgroundColor: "azure",
  },
  buttonOptions: {
    flexDirection: "row",
    justifyContent: "space-around",
    width: "92%",
  },
  buttonOptionsText: {
    margin: 5,
  },
  image: {
    height:20,
    width:14,
    marginHorizontal:3
  },
  imageText: {
    flexDirection: "row",
    marginLeft:'7%',
    marginTop:27
  },
  touchable: {
    flexDirection: "column",
    alignItems: "center",
  },
  subHeader: {
    fontFamily: "LatoRegular",
    color: "#747474",
    size: 12,
    lineHeight: 17.04
  },
});

export default GigMap;
