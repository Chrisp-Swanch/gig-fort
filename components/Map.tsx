import { FC,useState,useMemo,useEffect } from "react";
import {
  StyleSheet,
  Text,
  View,
  Image,
  ImageBackground,
  TouchableOpacity,
  Platform,
  Dimensions,
  Button
} from "react-native";
import MapView from "react-native-maps";
import { Marker } from "react-native-maps";
import { mapStyle } from "../util/mapStyle";
import { useGigs } from "../hooks/useGigs";
import { AntDesign,FontAwesome5,Ionicons } from "@expo/vector-icons";
import { format,isSameDay } from "date-fns";
import { mapProps } from "../routes/homeStack";
import { PROVIDER_GOOGLE } from "react-native-maps";
import * as Location from 'expo-location';
import icon from "../assets/blue_transparent_2.png";
import Carousel from "./Carousel";



type MapScreenNavgationProp = mapProps['navigation']

interface Props {
    navigation: MapScreenNavgationProp
}

const GigMap:FC<Props> = ({ navigation }):JSX.Element => {
  const [selectedDate, setSelectedDate] = useState(new Date());
  const [location, setLocation] = useState(null);
  const [errorMsg, setErrorMsg] = useState(null);
  const gigs = useGigs();


  const currentDay:string = useMemo(() => {
    const formattedDay = format(selectedDate,'EEEE')
    return formattedDay
  },[selectedDate])

  const currentWeek:string = useMemo(() => {
    const formattedDay = format(selectedDate,'LLLL do Y')
    return formattedDay
  },[selectedDate])

  //Filtering through gigs to return only current day's gigs
 
  const gigsToday = gigs?.filter((gig) => {
    const gigDate = new Date(gig.dateAndTime.seconds * 1000)
    return isSameDay(gigDate, selectedDate);
  });

  


  useEffect(() => {
    (async () => {
      
      let { status } = await Location.requestForegroundPermissionsAsync();
      if (status !== 'granted') {
        setErrorMsg('Permission to access location was denied');
        return;
      }

      let location = await Location.getCurrentPositionAsync({});
      setLocation(location);
    })();
  }, []);

  let text = 'Waiting..';
  if (errorMsg) {
    text = errorMsg;
  } else if (location) {
    text = JSON.stringify(location);
  }

  let userMarker = null;
  if (location) {
    userMarker = (
      <Marker
        coordinate={{
          latitude: location.coords.latitude,
          longitude: location.coords.longitude,
        }}
        title="You are here"
        icon = {icon}
        anchor={{ x: 0.5, y: 0.5 }} 
      />
    );
  }


  return (
    <View style={styles.container}>
      
      <View style = {styles.mapElements}>

        <TouchableOpacity style = {styles.mapElements_container}>
          <View>
            <Text style={styles.headerText_main}>{currentDay}</Text>
            <Text style={styles.headerText_sub}>{currentWeek}</Text>
          </View>
        </TouchableOpacity>

      </View>

      <View style={styles.mapContainer}>
        <MapView
          initialRegion={{
            latitude: -41.29416,
            longitude: 174.77782,
            latitudeDelta: 0.03,
            longitudeDelta: 0.03,
          }}
          style={styles.map}
          customMapStyle={mapStyle}
          provider = {PROVIDER_GOOGLE}
        >
          {userMarker}
          {gigsToday?.map((gig, i) => {
            return (
              <Marker
                key={i}
                anchor={{ x: 0.5, y: 0.5 }} 
                coordinate={{
                  latitude: gig.location.latitude,
                  longitude: gig.location.longitude,
                }}
                onPress={() => {
                  navigation.navigate("GigDetails", {
                    venue: gig.venue,
                    gigName: gig.gigName,
                    image: gig.image,
                    blurb: gig.blurb,
                    isFree: gig.isFree,
                    genre: gig.genre,
                    dateAndTime: { ...gig.dateAndTime },
                    ticketPrice: gig.ticketPrice,
                    tickets: gig.tickets,
                    address: gig.address,
                    links: gig.links,
                    gigName_subHeader: gig.gigName_subHeader
                  });
                }}
              >
                <View style={{ alignItems: 'center' }}>
                <Image style={styles.imageMain} source={require('../assets/map-pin-new.png')}/>
                  <Text style = {styles.markerText}>{gig.venue.length > 10 ? `${gig.venue.substring(0,10)}...` : gig.venue}</Text>
                </View>

              </Marker>
            );
          })}
        </MapView>
      </View>

      <Carousel setSelectedDate = {setSelectedDate}/>

    </View>
  );
};



const {width:screenWidth, height:screenHeight} = Dimensions.get('window')
const mapWidth = screenWidth * 0.9 //this sets width to 90%
const mapHeight = mapWidth /0.91 //this set height  based on the figma map aspect ratio of 0.91



const styles = StyleSheet.create({
  container: {
    // flexDirection: "column",
    // alignItems: "center",
    flex:1,
  },
  map: {
    height: '100%',
    width: '100%',
    flex:1,
    ...Platform.select({
      ios: {
        borderRadius:26,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.3,
        shadowRadius: 2,
      },
      android: {
        overflow: 'hidden',
        borderRadius:26,
        elevation: 4,
      }
    })
  },
  marker: {
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
  markerText: {
    color: 'black',
    fontWeight: 'bold',
    fontSize: 14,
    fontFamily:'LatoRegular',
    textShadowColor: 'white', // Outline color
    textShadowOffset: { width: 2, height: 2 }, // Outline thickness
    textShadowRadius: 3, // Outline blur radius
  },
  mapContainer:{
    marginTop: 0,
    // marginHorizontal: 20,
    // width: mapWidth,
    // height: mapHeight,
    flex:1,
    ...Platform.select({
      ios: {
        // borderRadius:26,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.3,
        shadowRadius: 2,
      },
      android: {
        overflow: 'hidden',
        // borderRadius:26,
        elevation: 4,
      }
    })
  },
  gigInfo: {
    // backgroundColor: '#68912b',
    // marginTop:20
  },
  gigInfo_text: {
    fontSize:10,
    color:'black',
    fontFamily:'NunitoSans',
    paddingTop: 25,
    textAlign:'center',
    fontWeight:'bold'
  },
  gigInfo_text_genre: {
    color: "white",
    fontFamily: "Helvetica-Neue",
    transform: [{ translateY: -5 }],
  },
  mapElements_container:{
    flexDirection:'row',
    backgroundColor: 'rgba(55, 125, 138,0.8)',
    alignSelf: 'flex-start',
    padding:'3%',
    borderRadius:8
  },
  headerText_main: {
    fontFamily: "NunitoSans",
    fontSize:25,
    lineHeight:34.1,
    color:'white',
    marginLeft:'7%'
  },
  headerText_sub: {
    fontFamily:'LatoRegular',
    size:14,
    lineHeight:16.8,
    color:'white',
    marginLeft:'7%'
  },
  callout: {
    width: "auto",
    height: "auto",
    backgroundColor: "azure",
  },
  buttonOptions: {
    flexDirection: "row",
    justifyContent: "space-between",
    width:'100%',
    position: 'absolute',
    top: '80%', 
    left: 0, 
    right: 0, 
    zIndex: 1,
    padding:'2%'
  },
  buttonOptionsText: {
    margin: 5,
  },
  image: {
    height:20,
    width:14,
    marginHorizontal:3
  },
  imageMain: {
    height:42,
    width:30,
    marginHorizontal:3, 

  },
  imageText: {
    flexDirection: "row",
    marginLeft:'7%',
    marginTop:27
  },
  touchable: {
    flexDirection: "column",
    alignItems: "center",
    // backgroundColor:'rgba(1,174,221, 0.9)',
    // padding:'2%',
    // width:'40%',
    // borderRadius:80
  },
  subHeader: {
    fontFamily: "LatoRegular",
    color: "#747474",
    size: 12,
    lineHeight: 17.04,
  },
  button:{
    flexDirection:'column',
    width:115,
    height:37,
    marginLeft:'7%',
    backgroundColor:'#377D8A',
    borderRadius:8,
    justifyContent:'center',
    marginTop:'6%'
  },
  buttonText: {
    color:'#FFFFFF',
    textAlign:'center',
    fontFamily: 'NunitoSans',
    fontSize:16,
    lineHeight:22
  },
  buttonAndSwitch:{
    flexDirection:'row',
    alignItems:'center',
    justifyContent:'space-between',
    marginBottom:'6%'
  },
  switch:{
    marginRight:'7%',
    zIndex:0,
  },
  switch_text: {
    fontFamily: 'LatoRegular',
    fontSize:10,
  },
  mapElements: {
    flexDirection: 'column',
    position: 'absolute',
    top: 10,
    left: 10,
    right: 0,
    zIndex: 1,
  },
  mapElements_top:{
    flex: 1,
  },
});

export default GigMap;