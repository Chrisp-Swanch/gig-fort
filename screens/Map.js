import { StyleSheet,View,Text,Pressable } from 'react-native'
import GigMap from '../components/GigMap'

const Map = ({ navigation }) => {
    return (
        <View style = {styles.container}>
            <GigMap navigation = {navigation}/>
            <View style = {styles.footer}>
            <Pressable
                title = "Go to list view"
                onPress = {() => navigation.navigate("List")}
                style = {styles.button}
            >
                <Text style = {styles.buttonText}>List View</Text>
            </Pressable>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex:1,
        backgroundColor: 'white',
        justifyContent: 'space-between'
    },
    footer:{
        backgroundColor: '#fff',
        height:50,
        justifyContent: 'center',
        alignItems:'center'
    },
    button: {
        backgroundColor: '#eee',
        padding:3,
        borderRadius:5
    },
    buttonText: {
        color:'black'
    }
})
 
export default Map;