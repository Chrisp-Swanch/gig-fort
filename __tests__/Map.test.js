import Map from '../screens/Map';
import GigMap from '../components/GigMap';
import * as React from 'react'
import { render,screen,fireEvent,waitFor, act } from '@testing-library/react-native'
import renderer from 'react-test-renderer'
import { format } from 'date-fns';

describe("Map component", () => {
  let navigation;

  beforeEach(() => {
    navigation = { navigate: jest.fn() };
  });

  test("should navigate to the List screen when the button is pressed", async () => {
    render(<Map navigation={navigation} />)
    const button = screen.getByText("List View")
    fireEvent.press(button);
    expect(navigation.navigate).toHaveBeenCalledWith("List");
  });

  // test("that UI stays consistent", async () => {
  //    act(async () => {
  //     const tree = await renderer.create(<Map />).toJSON();
  //     expect(tree).toMatchSnapshot();
  //   })
  // });

  test("that map header displays current date", () => {
    const currentDate = format(new Date(Date.now()),'EEE LLL do Y')
    const headerContent = `Gigs on ${currentDate}`
    render(<Map/>)
    const header = screen.getByText(headerContent)
    expect(header).toBeDefined()
  })

});