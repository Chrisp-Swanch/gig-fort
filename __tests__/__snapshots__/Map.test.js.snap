// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component that UI stays consistent 1`] = `
<View
  style={
    {
      "backgroundColor": "#f0ffff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontFamily": "Sofia-Pro",
          "fontSize": 25,
          "marginBottom": 10,
          "marginTop": 5,
        }
      }
      testID="gigMapHeader"
    >
      Gigs on Wed Feb 1st 2023
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#778899",
            "fontFamily": "Helvetica-Neue",
            "fontStyle": "italic",
          }
        }
      >
        Tap on
      </Text>
      <Image
        source={1}
        style={
          {
            "height": 20,
            "margin": 0,
            "paddingBottom": 10,
            "width": 20,
          }
        }
      />
      <Text
        style={
          {
            "color": "#778899",
            "fontFamily": "Helvetica-Neue",
            "fontStyle": "italic",
          }
        }
      >
         to see gig info
      </Text>
    </View>
    <AIRMap
      customMapStyle={
        [
          {
            "elementType": "geometry",
            "stylers": [
              {
                "color": "#1d2c4d",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "stylers": [
              {
                "color": "#8ec3b9",
              },
            ],
          },
          {
            "elementType": "labels.text.stroke",
            "stylers": [
              {
                "color": "#1a3646",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "administrative",
            "stylers": [
              {
                "visibility": "off",
              },
            ],
          },
          {
            "elementType": "geometry.stroke",
            "featureType": "administrative.country",
            "stylers": [
              {
                "color": "#4b6878",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "administrative.land_parcel",
            "stylers": [
              {
                "color": "#64779e",
              },
            ],
          },
          {
            "elementType": "geometry.stroke",
            "featureType": "administrative.province",
            "stylers": [
              {
                "color": "#4b6878",
              },
            ],
          },
          {
            "elementType": "geometry.stroke",
            "featureType": "landscape.man_made",
            "stylers": [
              {
                "color": "#334e87",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "landscape.natural",
            "stylers": [
              {
                "color": "#023e58",
              },
            ],
          },
          {
            "featureType": "poi",
            "stylers": [
              {
                "visibility": "off",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "poi",
            "stylers": [
              {
                "color": "#283d6a",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "poi",
            "stylers": [
              {
                "color": "#6f9ba5",
              },
            ],
          },
          {
            "elementType": "labels.text.stroke",
            "featureType": "poi",
            "stylers": [
              {
                "color": "#1d2c4d",
              },
            ],
          },
          {
            "elementType": "geometry.fill",
            "featureType": "poi.park",
            "stylers": [
              {
                "color": "#023e58",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "poi.park",
            "stylers": [
              {
                "color": "#3C7680",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "road",
            "stylers": [
              {
                "color": "#304a7d",
              },
            ],
          },
          {
            "elementType": "labels.icon",
            "featureType": "road",
            "stylers": [
              {
                "visibility": "off",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "road",
            "stylers": [
              {
                "color": "#98a5be",
              },
            ],
          },
          {
            "elementType": "labels.text.stroke",
            "featureType": "road",
            "stylers": [
              {
                "color": "#1d2c4d",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "road.highway",
            "stylers": [
              {
                "color": "#2c6675",
              },
            ],
          },
          {
            "elementType": "geometry.stroke",
            "featureType": "road.highway",
            "stylers": [
              {
                "color": "#255763",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "road.highway",
            "stylers": [
              {
                "color": "#b0d5ce",
              },
            ],
          },
          {
            "elementType": "labels.text.stroke",
            "featureType": "road.highway",
            "stylers": [
              {
                "color": "#023e58",
              },
            ],
          },
          {
            "featureType": "transit",
            "stylers": [
              {
                "visibility": "off",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "transit",
            "stylers": [
              {
                "color": "#98a5be",
              },
            ],
          },
          {
            "elementType": "labels.text.stroke",
            "featureType": "transit",
            "stylers": [
              {
                "color": "#1d2c4d",
              },
            ],
          },
          {
            "elementType": "geometry.fill",
            "featureType": "transit.line",
            "stylers": [
              {
                "color": "#283d6a",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "transit.station",
            "stylers": [
              {
                "color": "#3a4762",
              },
            ],
          },
          {
            "elementType": "geometry",
            "featureType": "water",
            "stylers": [
              {
                "color": "#0e1626",
              },
            ],
          },
          {
            "elementType": "labels.text.fill",
            "featureType": "water",
            "stylers": [
              {
                "color": "#4e6d70",
              },
            ],
          },
        ]
      }
      customMapStyleString="[{"elementType":"geometry","stylers":[{"color":"#1d2c4d"}]},{"elementType":"labels.text.fill","stylers":[{"color":"#8ec3b9"}]},{"elementType":"labels.text.stroke","stylers":[{"color":"#1a3646"}]},{"featureType":"administrative","elementType":"geometry","stylers":[{"visibility":"off"}]},{"featureType":"administrative.country","elementType":"geometry.stroke","stylers":[{"color":"#4b6878"}]},{"featureType":"administrative.land_parcel","elementType":"labels.text.fill","stylers":[{"color":"#64779e"}]},{"featureType":"administrative.province","elementType":"geometry.stroke","stylers":[{"color":"#4b6878"}]},{"featureType":"landscape.man_made","elementType":"geometry.stroke","stylers":[{"color":"#334e87"}]},{"featureType":"landscape.natural","elementType":"geometry","stylers":[{"color":"#023e58"}]},{"featureType":"poi","stylers":[{"visibility":"off"}]},{"featureType":"poi","elementType":"geometry","stylers":[{"color":"#283d6a"}]},{"featureType":"poi","elementType":"labels.text.fill","stylers":[{"color":"#6f9ba5"}]},{"featureType":"poi","elementType":"labels.text.stroke","stylers":[{"color":"#1d2c4d"}]},{"featureType":"poi.park","elementType":"geometry.fill","stylers":[{"color":"#023e58"}]},{"featureType":"poi.park","elementType":"labels.text.fill","stylers":[{"color":"#3C7680"}]},{"featureType":"road","elementType":"geometry","stylers":[{"color":"#304a7d"}]},{"featureType":"road","elementType":"labels.icon","stylers":[{"visibility":"off"}]},{"featureType":"road","elementType":"labels.text.fill","stylers":[{"color":"#98a5be"}]},{"featureType":"road","elementType":"labels.text.stroke","stylers":[{"color":"#1d2c4d"}]},{"featureType":"road.highway","elementType":"geometry","stylers":[{"color":"#2c6675"}]},{"featureType":"road.highway","elementType":"geometry.stroke","stylers":[{"color":"#255763"}]},{"featureType":"road.highway","elementType":"labels.text.fill","stylers":[{"color":"#b0d5ce"}]},{"featureType":"road.highway","elementType":"labels.text.stroke","stylers":[{"color":"#023e58"}]},{"featureType":"transit","stylers":[{"visibility":"off"}]},{"featureType":"transit","elementType":"labels.text.fill","stylers":[{"color":"#98a5be"}]},{"featureType":"transit","elementType":"labels.text.stroke","stylers":[{"color":"#1d2c4d"}]},{"featureType":"transit.line","elementType":"geometry.fill","stylers":[{"color":"#283d6a"}]},{"featureType":"transit.station","elementType":"geometry","stylers":[{"color":"#3a4762"}]},{"featureType":"water","elementType":"geometry","stylers":[{"color":"#0e1626"}]},{"featureType":"water","elementType":"labels.text.fill","stylers":[{"color":"#4e6d70"}]}]"
      initialRegion={
        {
          "latitude": -41.29416,
          "latitudeDelta": 0.03,
          "longitude": 174.77782,
          "longitudeDelta": 0.03,
        }
      }
      onChange={[Function]}
      onMapReady={[Function]}
      region={null}
      style={
        {
          "height": 450,
          "margin": 10,
          "width": "100%",
        }
      }
    />
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "width": "92%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#778899",
                "fontSize": 36,
              },
              undefined,
              {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#778899",
              "fontFamily": "Helvetica-Neue",
            }
          }
        >
          Previous day
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#778899",
                "fontSize": 36,
              },
              undefined,
              {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            {
              "color": "#778899",
              "fontFamily": "Helvetica-Neue",
            }
          }
        >
          Next day
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#68912b",
          "borderRadius": 5,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Helvetica-Neue",
          }
        }
      >
        List View
      </Text>
    </View>
  </View>
</View>
`;
